base/MAPL_EASEGridFactory.F90

```diff
@@ -0,0 +1,1866 @@
+            ic_1=i_1
+            ic_n=i_n
+         else
+            ic_1=i_1
+            if (i_n == this%im_world) then
+               ic_n=i_n+1
+            else
+               ic_n=i_n
+            end if
+         end if
+
+         jc_1=j_1
+         if (j_n == this%jm_world) then
+            jc_n=j_n+1
+         else
+            jc_n=j_n
+         end if
+
+         ! First we handle longitudes:
+         call ESMF_GridGetCoord(grid, coordDim=1, localDE=0, &
+              staggerloc=ESMF_STAGGERLOC_CENTER, &
+              farrayPtr=centers, _RC)
+         call ESMF_GridGetCoord(grid, coordDim=1, localDE=0, &
+              staggerloc=ESMF_STAGGERLOC_CORNER, &
+              farrayPtr=corners, _RC)
+         do j = 1, size(centers,2)
+            centers(:,j) = this%lon_centers(i_1:i_n)
+         end do
+         do j = 1, size(corners,2)
+            corners(:,j) = this%lon_corners(ic_1:ic_n)
+         end do
+
+         ! Now latitudes
+         call ESMF_GridGetCoord(grid, coordDim=2, localDE=0, &
+              staggerloc=ESMF_STAGGERLOC_CENTER, &
+              farrayPtr=centers, _RC)
+         call ESMF_GridGetCoord(grid, coordDim=2, localDE=0, &
+              staggerloc=ESMF_STAGGERLOC_CORNER, &
+              farrayPtr=corners, _RC)
+
+         do i = 1, size(centers,1)
+            centers(i,:) = this%lat_centers(j_1:j_n)
+         end do
+         do i = 1, size(corners,1)
+            corners(i,:) = this%lat_corners(jc_1:jc_n)
+         end do
+      end if
+
+      _RETURN(_SUCCESS)
+
+   end subroutine add_horz_coordinates
+
+
+   subroutine initialize_from_file_metadata(this, file_metadata, unusable, force_file_coordinates, rc)
+      use MAPL_KeywordEnforcerMod
+      use MAPL_BaseMod, only: MAPL_DecomposeDim
+
+      class (EASEGridFactory), intent(inout)  :: this
+      type (FileMetadata), target, intent(in) :: file_metadata
+      class (KeywordEnforcer), optional, intent(in) :: unusable
+      logical, optional, intent(in) :: force_file_coordinates
+      integer, optional, intent(out) :: rc
+
+      integer :: status
+
+      class (CoordinateVariable), pointer :: v
+      class (*), pointer :: ptr(:)
+
+      character(:), allocatable :: lon_name
+      character(:), allocatable :: lat_name
+      character(:), allocatable :: lev_name
+      character(:), allocatable :: grid_name
+      integer :: i
+      logical :: hasLon, hasLat, hasLongitude, hasLatitude, hasLev,hasLevel,regLat,regLon
+      real(kind=REAL64) :: del12,delij
+
+      integer :: i_min, i_max
+      real(kind=REAL64) :: d_lat, d_lat_temp, extrap_lat
+      logical :: is_valid, use_file_coords, compute_lons, compute_lats, has_bnds
+
+      _UNUSED_DUMMY(unusable)
+
+      if (present(force_file_coordinates)) then
+         use_file_coords = force_file_Coordinates
+      else
+         use_file_coords = .false.
+      end if
+
+      ! Cannot assume that lats and lons are evenly spaced
+      this%is_regular = .false.
```

Comment:
"is_regular" may be misleading for the EASE grid.  In my understanding, the EASE grid is a "regular" grid.  It does not have even lat spacing, but it's still a regular grid (as opposed to, say, the cube-sphere tile space).    Maybe "is_evenspaced" (or something to that effect) would be a less confusing variable name.